.storybook/preview.js

```diff
@@ -23,139 +18,215 @@ Object.entries(PrimerBreakpoints).forEach(([viewport, value]) => {
+          if (positionA === positionB) {
+            return valueA.localeCompare(valueB)
+          }
 
-        // If the positions are the same, sort them alphabetically
-        if (positionA === positionB) {
-          return valueA.localeCompare(valueB)
+          // Otherwise, compare by position in the given order
+          return positionA - positionB
         }
 
-        // Otherwise, compare by position in the given order
-        return positionA - positionB
-      }
-
-      function getHierarchy(story) {
-        const hierarchy = story.title.split('/')
-        hierarchy.push(story.name)
-        return hierarchy
-      }
+        function getHierarchy(story) {
+          const hierarchy = story.title.split('/')
+          hierarchy.push(story.name)
+          return hierarchy
+        }
 
-      return compare(getHierarchy(a), getHierarchy(b))
+        return compare(getHierarchy(a), getHierarchy(b))
+      },
+    },
+    viewport: {
+      viewports: {
+        ...storybookViewports,
+      },
     },
   },
-  viewport: {
-    viewports: {
-      ...storybookViewports,
+}
+
+const primerThemes = [
+  {value: 'light', left: '☀️', title: 'Light'},
+  {value: 'light_colorblind', left: '☀️', title: 'Light Protanopia & Deuteranopia'},
+  {value: 'light_tritanopia', left: '☀️', title: 'Light Tritanopia'},
+  {value: 'light_high_contrast', left: '☀️', title: 'Light High Contrast'},
+  {value: 'dark', left: '🌗', title: 'Dark'},
+  {value: 'dark_dimmed', left: '🌗', title: 'Dark Dimmed'},
+  {value: 'dark_colorblind', left: '🌗', title: 'Dark Protanopia & Deuteranopia'},
+  {value: 'dark_tritanopia', left: '🌗', title: 'Dark Tritanopia'},
+  {value: 'dark_high_contrast', left: '🌗', title: 'Dark High Contrast'},
+]
+
+export const globalTypes = {
+  colorScheme: {
+    name: 'Theme',
+    description: 'Switch themes',
+    defaultValue: 'light',
+    toolbar: {
+      icon: 'contrast',
+      items: [...primerThemes, {value: 'all', left: '', title: 'All'}],
+      showName: true,
+      dynamicTitle: true,
     },
+    showSurroundingElements: {},
   },
 }
+
+export const decorators = [
+  (Story, context) => {
+    document.body.setAttribute('data-color-mode', context.globals.colorScheme.startsWith('light') ? 'light' : 'dark')
+    document.body.setAttribute(
+      'data-light-theme',
+      context.globals.colorScheme.startsWith('light') ? context.globals.colorScheme : undefined,
+    )
+    document.body.setAttribute(
+      'data-dark-theme',
+      context.globals.colorScheme.startsWith('dark') ? context.globals.colorScheme : undefined,
+    )
+    const showSurroundingElements =
+      context.globals.showSurroundingElements ?? window.localStorage.getItem('showSurroundingElements') === 'true'
+    return context.globals.colorScheme === 'all' ? (
+      primerThemes.map(({value: theme}) => (
+        <ThemeProvider dayScheme={theme} nightScheme={theme} colorMode="day">
+          <div
+            key={theme}
```

Comment:
Noticed that `key` was on the `div` versus the top-level element `ThemeProvider`

```suggestion
        <ThemeProvider key={theme} dayScheme={theme} nightScheme={theme} colorMode="day">
          <div
            
```